bundles/org.openhab.binding.mqtt.awtrixlight/src/main/java/org/openhab/binding/mqtt/awtrixlight/internal/handler/AwtrixLightAppHandler.java

```diff
@@ -0,0 +1,653 @@
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.util.ColorUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AwtrixLightAppHandler} is responsible for handling commands for an app and will send mqtt messages to
+ * update the app configuration on the awtrix clock. It will also emit trigger events as long as the app is locked on
+ * the clock.
+ *
+ * @author Thomas Lauterbach - Initial contribution
+ */
+@NonNullByDefault
+public class AwtrixLightAppHandler extends BaseThingHandler implements MqttMessageSubscriber {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Set.of(THING_TYPE_APP);
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private String channelPrefix = "";
+    private String appName = "";
+    private Boolean synchronizationRequired = true;
+    private Boolean buttonControlled = false;
+    private Boolean active = true;
+
+    private AwtrixApp app = new AwtrixApp();
+
+    private final Object syncLock = new Object();
+    private @Nullable ScheduledFuture<?> finishInitJob;
+
+    public AwtrixLightAppHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Received command {} of type {} on channel {}", command.toString(), command.getClass(),
+                channelUID.getAsString());
+        if (this.synchronizationRequired) {
+            // Don't accept any commands while we're synchronizing our settings
+            return;
+        }
+
+        if (command instanceof RefreshType) {
+            updateApp();
+            return;
+        }
+        switch (channelUID.getId()) {
+            case CHANNEL_ACTIVE:
+                // WARNING: Inactive Apps will return with default values after OH reboot
+                if (command instanceof OnOffType) {
+                    if (OnOffType.OFF.equals(command)) {
+                        this.active = false;
+                        deleteApp();
+                    } else if (OnOffType.ON.equals(command)) {
+                        this.active = true;
+                    }
+                }
+                break;
+            case CHANNEL_RESET:
+                if (command instanceof OnOffType) {
+                    if (OnOffType.ON.equals((OnOffType) command)) {
+                        deleteApp();
+                        this.app = new AwtrixApp();
+                        updateApp();
+                        initStates();
+                        return;
+                    }
+                }
+                break;
+            case CHANNEL_COLOR:
+                if (command instanceof HSBType) {
+                    this.app.setColor(ColorUtil.hsbToRgb((HSBType) command));
```

Comment:
Please check all instanceof for pattern matching preventing doubvle cast.

```suggestion
                if (command instanceof HSBType hsbCommand) {
                    this.app.setColor(ColorUtil.hsbToRgb(hsbCommand));
```